tm2/pkg/bft/rpc/core/blocks.go

```diff
@@ -417,11 +417,11 @@ func BlockResults(ctx *rpctypes.Context, heightPtr *int64) (*ctypes.ResultBlockR
 	return res, nil
 }
 
-func getHeight(currentHeight int64, heightPtr *int64) (int64, error) {
+func getHeight(currentHeight int64, heightPtr *int64, min int64) (int64, error) {
 	if heightPtr != nil {
 		height := *heightPtr
-		if height <= 0 {
-			return 0, fmt.Errorf("height must be greater than 0")
+		if height < min {
+			return 0, fmt.Errorf("height must be greater than or equal to %d", min)
```

Comment:
the error and the if expression are not the same:

```suggestion
		if height <= min {
			return 0, fmt.Errorf("height must be greater than or equal to %d", min)
```

Reply:
shall we add a test for this corner case?

Reply:
mmh, the check and error are matching, the error message must be the inverse of the if expression

if `height` is less than `min`, then we stop and return an error saying that `height` must be greater than or equal to `min`

it could be rewritten as `if !(height >= min) {` if you think it's clearer

I will add a test

Reply:
test added in bad682a272123a943fd2163cb501dbadc837500b